scipy/stats/_qmc.py

```diff
@@ -860,15 +968,41 @@ class LatinHypercube(QMCEngine):
            [6.80338249, 3.08795949],
            [2.65448791, 3.83491828]])
 
+    Use the `optimization` keyword argument to produce a LHS with
+    lower discrepancy at higher computational cost.
+
+    >>> sampler = qmc.LatinHypercube(d=2, optimization="random-cd")
+    >>> sample = sampler.random(n=5)
+    >>> qmc.discrepancy(sample)
+    0.0176...  # random
+
     """
 
     def __init__(
         self, d: IntNumber, *, centered: bool = False,
+        optimization: Optional[Literal["random-cd"]] = None,
         seed: SeedType = None
     ) -> None:
         super().__init__(d=d, seed=seed)
         self.centered = centered
 
+        lhs_methods: Dict[Optional[Literal["random-cd"]], Callable] = {
+            None: self._random,
+            "random-cd": self._random_cd,
+        }
+
+        try:
+            if optimization is not None:
+                optimization = optimization.lower()
+            self.lhs_method = lhs_methods[optimization]
```

Comment:
@BvB93 can you help here? I am not sure what's best to do...

`scipy/stats/_qmc.py:996: error: Incompatible types in assignment (expression has type "str", variable has type "Optional[Literal['random-cd']]")  [assignment]`

Reply:
I think this might be a use-case for a `# type: ignore[assignment]` comment.

In my experience mypy's support for string literals is not that great; 
thus you're getting an error as mypy thinks the `str.lower` call returns a normal string instead of the appropiate string-literal.

Reply:
Ok then I will use this. Thanks! It's quite annoying that it is not well supported...